From 6ffd142b909ec6fb8b50458a2ed536ae6e9ffaf9 Mon Sep 17 00:00:00 2001
From: NRottmann <Nils.Rottmann@tuhh.de>
Date: Thu, 6 Jul 2017 12:32:29 +0200
Subject: [PATCH] Changing positioning of HippoCampus and of the IMU

The position of the IMU relative to the HippoCampus has been changed in
order to get exact orientation data in NED frame from the control state topic.

The placement of the HippoCampus at the beginning of the Simulation has been changed
in oder to start at zero rotation in reagrd to NED frame.
---
 models/hippocampus/hippocampus.sdf | 13 ++++++++-----
 worlds/underwater.world            |  8 ++++++--
 2 files changed, 14 insertions(+), 7 deletions(-)

diff --git a/models/hippocampus/hippocampus.sdf b/models/hippocampus/hippocampus.sdf
index 5bd0233864..dd1930b8b4 100644
--- a/models/hippocampus/hippocampus.sdf
+++ b/models/hippocampus/hippocampus.sdf
@@ -7,9 +7,9 @@
   -->
   <model name='hippocampus'>
     <link name='base_link'>
-      <pose frame=''>0 0 0 0 0 0</pose>
+      <pose frame=''>0 0 0 0 -0 0</pose>
       <inertial>
-        <pose frame=''>0 0 0 0 0 0</pose>
+        <pose frame=''>0 0 0 0 -0 0</pose>
         <!--
           These values are a compromise between the dry mass and added mass
           values used to correct for hydrodynamic effects.
@@ -167,12 +167,15 @@
 
     <!--
         A new link is defined which hold the IMU (e.g. acceleration sensors, ...). For the simulation it is positioned
-        at the center of the Hippocampus
+        at the center of the Hippocampus.
+        The orientation of the IMU is chosen in regard to the orientation of the Body coordinate system of the
+        HippoCampus. PX4 uses the data from the IMU to calculate the orientation. Therefore PX4 assumes a NED (North-
+        East-Down) coordinate frame whereas Gazebo uses a ENU (East-North-Up) coordinate frame.
     -->
     <link name='hippocampus/imu_link'>
-      <pose frame=''>0 0 0 0 0 1.570796</pose>
+      <pose frame=''>0 0 0 3.1415 0 0</pose>
       <inertial>
-        <pose frame=''>0 0 0 0 0 1.570796</pose>
+        <pose frame=''>0 0 0 0 0 0</pose>
         <mass>0.015</mass>
         <inertia>
           <ixx>1e-05</ixx>
diff --git a/worlds/underwater.world b/worlds/underwater.world
index 5ec2735d5a..0745f47774 100644
--- a/worlds/underwater.world
+++ b/worlds/underwater.world
@@ -13,10 +13,14 @@
       <uri>model://sun</uri>
     </include-->
 
-    <!-- Include the model of the Hippocampus from TUHH.-->
+    <!--
+        Include the model of the Hippocampus from TUHH.
+        The pose is chosen that the hippocampus starts at position (0,0,0) with its own coordinate frame equal to the
+        NED World frame.
+    -->
     <include>
       <uri>model://hippocampus</uri>
-      <pose>0 0 0 0 0 0</pose>
+      <pose>0 0 0 0 3.1415 -1.57075</pose>
     </include>
 
     <!-- Add physics -->
